Dev-Corner/tutorials/files/hands_on-visualizations/alignment_rname_boxplot/templates/alignment_rname_boxplot.mako

```diff
@@ -14,15 +16,28 @@
     ## Actual file URL:
     file_url = os.path.join(root, 'datasets', hdadict['id'], "display?to_ext="+hda.ext)
 
-    ## Ensure BAI index is symlinked
-    bai_target = hda.file_name+'.bai'
+    # File path in tmp dir:
+    temp_data_dir = tempfile.gettempdir()+'/plugins/visualizations/'+visualization_name
+    alignment_file = temp_data_dir + '/' + hdadict['id'] + '_' + os.path.basename(hda.file_name)
+    bai_target = alignment_file + '.bai'
+
+    ## Ensure temp dir exists
+    if not os.path.exists(temp_data_dir):
+        os.makedirs(temp_data_dir)
+
+    ## Make symlink to datafile in tmp dir
+    if not os.path.isfile(alignment_file):
+        os.symlink(hda.file_name , alignment_file)
 
+    ## Make symlink to datafile in temp dir (to keep Galaxy storage dir clean)
     if not os.path.isfile(bai_target):
-        os.symlink(hda.metadata.bam_index.file_name, bai_target)
+        if not os.path.isfile(hda.metadata.bam_index.file_name):
```

Comment:
Haven't seen this but wasn't sure if This was 100% taken care of by the framework or bij tools themselves. Thanks again for The review. I Will make sure it is fixed in time